bsd/models.py

```diff
@@ -1,10 +1,316 @@
+
+"""Get BSD api"""
+bsdApi = BSD().api
 
 
 class BSDProfile(models.Model):
     # 0 should only be used for legacy records that predate this field
     cons_id_default = '0'
     cons_id = models.CharField(default=cons_id_default, max_length=128)
     user = models.OneToOneField(User, on_delete=models.CASCADE)
+
+
+class BSDEventManager(models.Manager):
+
+    def from_json(self, data):
+
+        """Assume duration type = minutes for BSD data"""
+        duration_type = 1
+
+        try:
+            """Expected format for get_event_details"""
+            capacity = data["days"][0]["capacity"]
+            duration_count = data["days"][0]["duration"]
+            start_datetime_utc = datetime.datetime.strptime(
+                data["days"][0]["start_dt"],
+                '%Y-%m-%d %H:%M:%S'
+            )
+            venue_state_or_territory = data["venue_state_code"]
+        except KeyError:
+            """Expected format for get_events_for_cons"""
+            capacity = data["venue_capacity"]
+            duration_count = data["duration"]
+            start_datetime_utc = datetime.datetime.strptime(
+                data["start_datetime_system"],
+                '%Y-%m-%d %H:%M:%S'
+            )
+            venue_state_or_territory = data["venue_state_cd"]
+
+        """Get Local Datetime"""
+        utc_zone = tz.gettz('UTC')
+        local_zone = tz.gettz(data["local_timezone"])
+        utc_datetime = start_datetime_utc.replace(tzinfo=utc_zone)
+        local_datetime = utc_datetime.astimezone(local_zone)
+
+        bsd_event = self.model(
+            capacity=capacity,
+            contact_phone=data["contact_phone"],
+            creator_cons_id=data["creator_cons_id"],
+            event_id_obfuscated=data["event_id_obfuscated"],
+            event_type=data["event_type_id"],
+            flag_approval=int(data["flag_approval"]),
+            host_name=data["creator_name"],
+            name=data["name"],
+            description=data["description"],
+            duration_count=duration_count,
+            duration_type=duration_type,
+            host_receive_rsvp_emails=data["host_receive_rsvp_emails"],
+            public_phone=data["public_phone"],
+            start_day=local_datetime.date(),
+            start_time=local_datetime.time(),
+            start_time_zone=data["local_timezone"],
+            venue_name=data["venue_name"],
+            venue_addr1=data["venue_addr1"],
+            venue_addr2=data["venue_addr2"],
+            venue_city=data["venue_city"],
+            venue_country=data["venue_country"],
+            venue_directions=data["venue_directions"],
+            venue_state_or_territory=venue_state_or_territory,
+            venue_zip=data["venue_zip"],
+        )
+        return bsd_event
+
+
+class BSDEvent(models.Model):
+    """BSD Event"""
+    objects = BSDEventManager()
+
+    duration_type_choices = (
+        (1, 'Minutes'),
+        (2, 'Hours'),
+    )
+    event_type_choices = (
```

Comment:
Not required for this ticket, but is there anyway we could make this not be hardcoded?

Reply:
we discussed. doesnt have to be hardcoded but this seems like preferred approach for now.